ios/FluentUI/People Picker/ContactView.swift

```diff
@@ -0,0 +1,172 @@
+    ///   - lastName: String that will be the text of the bottom label
+    public init(avatarView: AvatarView, firstName: String, lastName: String) {
+        self.avatarView = avatarView
+        self.firstNameLabel = UILabel()
+        self.lastNameLabel = UILabel()
+        super.init(frame: .zero)
+        self.backgroundColor = Colors.surfacePrimary
+        self.translatesAutoresizingMaskIntoConstraints = false
+        setupFirstNameLabel(using: firstName)
+        setupLastNameLabel(using: lastName)
+        setupLayout()
+    }
+
+    /// Initializes the contact view with an avatar view and an identifier
+    ///
+    /// - Parameters:
+    ///   - avatarView: The AvatarView that will be displayed within the ContactView
+    ///   - identifier: String that will be used to identify the contact (e.g. email, phone number, first name)
+    public init(avatarView: AvatarView, identifier: String) {
+        self.avatarView = avatarView
+        super.init(frame: .zero)
+        self.backgroundColor = Colors.surfacePrimary
+        self.translatesAutoresizingMaskIntoConstraints = false
+        setupIdentifierLabel(using: identifier)
+        setupLayout()
+    }
+
+    public required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    private func setupLayout() {
+        var constraints = [NSLayoutConstraint]()
+        constraints.append(self.widthAnchor.constraint(equalToConstant: Constants.contactWidth))
+        constraints.append(contentsOf: avatarLayoutConstraints())
+
+        if firstNameLabel != nil && lastNameLabel != nil {
+            constraints.append(contentsOf: firstNameLayoutConstraints())
+            constraints.append(contentsOf: lastNameLayoutConstraints())
+        } else {
+            constraints.append(contentsOf: identifierLayoutConstraints())
+        }
+
+        NSLayoutConstraint.activate(constraints)
+    }
+
+    private func avatarLayoutConstraints() -> [NSLayoutConstraint] {
+        addSubview(avatarView)
+        avatarView.translatesAutoresizingMaskIntoConstraints = false
+
+        var constraints = [NSLayoutConstraint]()
+        constraints.append(avatarView.heightAnchor.constraint(equalToConstant: Constants.avatarViewHeight))
+        constraints.append(avatarView.widthAnchor.constraint(equalToConstant: Constants.avatarViewWidth))
+        constraints.append(avatarView.centerXAnchor.constraint(equalTo: self.centerXAnchor))
+        constraints.append(avatarView.topAnchor.constraint(equalTo: self.topAnchor))
+        constraints.append(avatarView.leadingAnchor.constraint(equalTo: self.leadingAnchor))
+        constraints.append(avatarView.trailingAnchor.constraint(equalTo: self.trailingAnchor))
+        return constraints
+    }
+
+    private func firstNameLayoutConstraints() -> [NSLayoutConstraint] {
+        var constraints = [NSLayoutConstraint]()
+        if let firstNameLabel = firstNameLabel {
```

Comment:
>if let firstNameLabel = firstNameLabel { [](start = 8, length = 40)

guard let firstNameLabel = firstNameLabel else {
       return nil
}

Then we can make contraints a constant (let)

Reply:
I understand what you mean by using `guard` in order to return and exit out of the function early. I'm a bit puzzled regarding making `constraints` a `let`. Since if it's a `let`, we won't be able to append the constraints to array one by one. Are you suggesting to initialize the array with all of the constraints like `let constraints = [constraint_1, constraint_2, ...]`?

Reply:
yes, that's what I'm suggesting